### ISSUE TITLE
image import failed after recent pantry API changes

### ISSUE THREAD

**Issue by davepacheco** [oxidecomputer/omicron#4595]
tl;dr: oxidecomputer/crucible#1039 changed the pantry API incompatibly to match our API guidelines, [replacing underscores with hyphens](https://github.com/oxidecomputer/crucible/pull/1039/files#diff-9ad5fc973e7f8816209382970985dd47b60f1d55b57aceff543e50a823a721d3R191).  As of Omicron "main" right now, the [Pantry implementation that we're pulling in](https://github.com/oxidecomputer/omicron/blob/92aed1a25a57d8483fc212ba245d871514aeadee/package-manifest.toml#L398) post-dates this change, but the [crucible-pantry-client implementation that we're pulling in](https://github.com/oxidecomputer/omicron/blob/92aed1a25a57d8483fc212ba245d871514aeadee/Cargo.toml#L175) predates it.  This breaks image import and I'm not sure what else.

---

How I got here: I tried to import an image and got:

```
dap@ivanova ~ $ oxide disk import --path debian-11-genericcloud-amd64.raw --disk debian-boo --project test-project --description debian --snapshot debian-import --image debian --image-description debian --image-os debian --image-version 11
[00:00:00] [███░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░] 194.00 MiB/2.00 GiB (783.64 MiB/s, 2s)sending chunk to thread failed with SendError { .. }
channel closed
```

(I filed oxidecomputer/oxide.rs#448 for having the CLI provide more detail on this failure mode.)

I went looking at one of the Nexus logs for message emitted when this happens and I found:

```
16:33:50.143Z INFO f45dcde1-e10d-440b-ade6-a9ab6f283652 (dropshot_external): request completed 
    error_message_external = Internal Server Error
    error_message_internal = error sending bulk write to pantry: Communication Error: error sending request for url (http://[fd00:1122:3344:102::5]:17000/crucible/pantry/0/volume/b6db3011-b1cd-43f5-a1e6-611248d65a00/bulk_write): error writing a body to connection: Broken pipe (os error 32)
    file = /home/build/.cargo/git/checkouts/dropshot-a4a923d29dccc492/ff87a01/dropshot/src/server.rs:841
    latency_us = 371589
    local_addr = 172.30.2.6:443
    method = POST 
    remote_addr = 172.20.2.70:46781
    req_id = 852eddaa-2464-4844-8211-6628a664495c
    response_code = 500
    uri = //v1/disks/debian-boo/bulk-write?project=test-project
```

Over in the pantry, I'm not sure which messages were relevant (I did not find any with that `req_id`) but I did find a bunch of messages like this for the same volume id:

```
16:33:49.922Z INFO crucible-pantry (dropshot): request completed
    error_message_external = Not Found
    error_message_internal = no route found (no path in router)
    latency_us = 24
    local_addr = [fd00:1122:3344:102::5]:17000
    method = POST
    remote_addr = [fd00:1122:3344:102::4]:42461
    req_id = fd8c1805-0f02-4961-99d7-3c48d9e33009
    response_code = 404
    uri = /crucible/pantry/0/volume/b6db3011-b1cd-43f5-a1e6-611248d65a00/bulk_write
```

These are dropshot reporting that there is no route handler for this route, which is pretty unusual.  That's when I went looking for a breaking API change and found the above.

**Comment by bnaecker** [comment]
Ugh, sorry about this one. I definitely just missed these dependencies in the top-level Cargo.toml.

**Comment by leftwo** [comment]
https://github.com/oxidecomputer/omicron/pull/4597

**Comment by leftwo** [comment]
The fix is in: https://github.com/oxidecomputer/omicron/pull/4597
